### PR TITLE
[TREE] Utility to enter a range of entries in TEntryList

### PR DESCRIPTION
**Pull request by vepadulano**
Add a utility function to call `TEntryList::Enter` with entries in a certain range, instead of having to do the loop manually. Especially useful in PyROOT to avoid doing the same in a Python loop
### Initial idea
```py
>>> import ROOT
>>> e = ROOT.TEntryList()
>>> e.GetN()
0
>>> ROOT.ROOT.Detail.EnterRange(e, 0, 10)
>>> e.GetN()
10
```

Not sure about the namespace and the naming, can be discussed

### Final interface

```py
>>> import ROOT
>>> e = ROOT.TEntryList()
>>> e.EnterRange(0,10)
>>> e.GetN()
10
```

### CODE REVIEW

**Review by eguiraud — APPROVED** [review]
Idea + implementation looks good to me.

Naming/placement: if we are putting the function in `TEntryList.h` anyway, I guess it can be a method. @pcanal what do you think?

Unrelated, but I hope you don't really need the double `ROOT.ROOT` to call this from PyROOT, I think that has been fixed years ago :smile:

**Comment by vepadulano** [comment]
> Unrelated, but I hope you don't really need the double ROOT.ROOT to call this from PyROOT, I think that has been fixed years ago

I thought so too, but currently
```py
>>> ROOT.Detail.EnterRange
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: <namespace cppyy.gbl.Detail at 0x56507072a2f0> has no attribute 'EnterRange'. Full details:
  type object 'Detail' has no attribute 'EnterRange'
  'Detail::EnterRange' is not a known C++ class
  'EnterRange' is not a known C++ template
  'EnterRange' is not a known C++ enum
```
So maybe it's something that must be "activated" somehow. I'd need to see how it's done with other classes/functions

**Comment by eguiraud** [comment]
It could be that it only works for classes and not for functions. Can you please open an issue about that? It's unrelated to the PR but I can't find a jira ticket about this thing.

**Comment by vepadulano** [comment]
>  Can you please open an issue about that?

See https://github.com/root-project/root/issues/8745

**Review by eguiraud — COMMENTED** [review]
Alright, looks good to me!

The feature needs a test before merging.

Also maybe I'd put the `tree` argument before the `step` argument because I don't expect the `step` argument to be used much? (but maybe that's also the case for `tree`?)

**Comment by vepadulano** [comment]
> Also maybe I'd put the tree argument before the step argument because I don't expect the step argument to be used much? (but maybe that's also the case for tree?)

I put the step first because I felt it went "logically" after the other two integer arguments. I really wouldn't know what could be used more between that and the tree argument

**Comment by vepadulano** [comment]
The test failures report the file created during the test does not exist, but this node ran the test fine https://lcgapp-services.cern.ch/root-jenkins/job/root-pullrequests-build/124359/testReport/projectroot.tree.tree/test/gtest_tree_tree_test_entrylist_enterrange/ 

Probably using the same filename for multiple tests leads to a recreation/deletion of the file that is then not usable by the new test?

File: tree/tree/test/entrylist_enterrange.cxx
Comment (eguiraud) [inline tree/tree/test/entrylist_enterrange.cxx:31]:
```suggestion
   const auto start1{10};
```

**Comment by vepadulano** [comment]
> Using the same file in multiple tests is never good, they might run concurrently

You are right, I changed the filename in the latest commit :+1:

**Review by eguiraud — APPROVED** [review]
LGTM

**Comment by vepadulano** [comment]
Added commit message and rebased. Will merge if tests pass

**Pull request merged**
